packages/mobile/src/exchange/ExchangeTradeScreen.tsx

```diff
@@ -130,75 +131,98 @@ export class ExchangeTradeScreen extends React.Component<Props, State> {
     navigate(Screens.ExchangeReview, {
-      exchangeInfo: { makerToken, makerAmount: parseInputAmount(makerTokenAmount) },
+      confirmationInput: { makerToken, makerAmount: parseInputAmount(makerTokenAmount) },
     })
   }
 
-  newBalance = (isDollar: boolean, takerTokenAmount: BigNumber) => {
+  getNewTakerBalance = (takerTokenAmount: BigNumber) => {
     const { dollarBalance, goldBalance } = this.props
-    return isDollar
+    return this.isDollarToGold()
       ? getNewTakerBalance(goldBalance, takerTokenAmount)
-      : CURRENCIES[Token.DOLLAR].symbol + getNewTakerBalance(dollarBalance, takerTokenAmount)
+      : getNewTakerBalance(dollarBalance, takerTokenAmount)
   }
 
   hasError = () => {
     return !!this.props.error
   }
 
   isExchangeInvalid = () => {
-    // convert to numeral.value so cases like null/undefined, or a string of zeros '0000' or empty string are all treated as invalid
-    const amount = new BigNumber(numeral(this.state.makerTokenAmount).value())
     const makerBalance = this.getMakerBalance()
-    const amountIsInvalid = amount.isGreaterThan(makerBalance) || amount.isLessThan(0.01)
-    const exchangeRateIsInvalid = getRateForMakerToken(
-      this.props.exchangeRatePair,
-      this.state.makerToken
-    ).isZero()
-    return amountIsInvalid || exchangeRateIsInvalid || this.hasError()
+    const amount = parseInputAmount(this.state.makerTokenAmount)
+    const amountIsInvalid =
+      amount.isGreaterThan(makerBalance) || amount.isLessThan(TRANSACTION_MIN_AMOUNT)
+    const exchangeRate = getRateForMakerToken(this.props.exchangeRatePair, this.state.makerToken)
+    const exchangeRateIsInvalid = exchangeRate.isLessThanOrEqualTo(0)
+    const takerToken = this.isDollarToGold() ? Tokens.GOLD : Tokens.DOLLAR
+    const takerAmountIsInvalid = getTakerAmount(
+      amount,
+      exchangeRate,
+      CURRENCIES[takerToken].displayDecimals
+    ).isLessThanOrEqualTo(0)
+
+    return amountIsInvalid || exchangeRateIsInvalid || takerAmountIsInvalid || this.hasError()
   }
 
   getMakerBalance = () => {
-    return new BigNumber((this.isDollar() ? this.props.dollarBalance : this.props.goldBalance) || 0)
-  }
-
-  getFormattedMakerBalance = () => {
-    return getMoneyDisplayValue(this.getMakerBalance())
+    return new BigNumber(
+      (this.isDollarToGold() ? this.props.dollarBalance : this.props.goldBalance) || 0
+    )
   }
 
-  isDollar = () => {
-    return this.state.makerToken === Token.DOLLAR
+  isDollarToGold = () => {
+    return this.state.makerToken === Tokens.DOLLAR
   }
 
-  inputValue = () => {
+  getInputValue = () => {
     if (this.state.makerTokenAmount) {
       return CURRENCIES[this.state.makerToken].symbol + this.state.makerTokenAmount
     } else {
       return ''
     }
   }
 
-  render() {
+  getMakerTakerProps = () => {
     const { t } = this.props
+    const dollarProps = {
+      token: Tokens.DOLLAR,
+      tokenText: t('global:celoDollars') + ' (cUSD)',
```

Comment:
We should avoid programmatically concatenating user facing strings. This makes it impossible to correctly translate this in languages that use a different order.

Reply:
Agreed. Not new logic though, just moved it around